core_types/src/chain_header.rs

```diff
@@ -109,25 +114,22 @@ impl ToEntry for ChainHeader {
     fn to_entry(&self) -> Entry {
         Entry::new(
             &EntryType::ChainHeader,
-            &self.to_json().expect("entry should be valid"),
+            &JsonString::from(self.to_owned()),
         )
     }
 
     fn from_entry(entry: &Entry) -> Self {
-        return ChainHeader::from_json_str(&entry.value())
-            .expect("entry is not a valid ChainHeader Entry");
+        ChainHeader::from(entry.value().to_owned())
     }
 }
 
 impl AddressableContent for ChainHeader {
     fn content(&self) -> Content {
-        self.to_json()
-            .expect("could not Jsonify ChainHeader as Content")
+        Content::from(self.to_owned())
```

Comment:
I'd prefer `self.to_json()` or `self.to_content()`
This kind of hides the fact that we are serializing to a JsonString.

Reply:
@ddd-mtl does it? isn't the point of `From` that it defines type conversions?

Reply:
i'll try `into()` as that seems pretty standard